### PR TITLE
This is a temporary hack to workaround the way we are currently lower…

### PR DESCRIPTION
**Pull request by schweitzpgi**
…ing calls. More work is needed here.

This is a quick workaround for LAPACK sources. It does not break our tests.

### CODE REVIEW

File: flang/lib/Lower/ConvertExpr.cpp [hunk 1]
```diff
@@ -933,7 +940,11 @@ class ExprLowering {
           addr = builder.createTemporary(getLoc(), val.getType());
           builder.create<fir::StoreOp>(getLoc(), val, addr);
         }
-        argTypes.push_back(addr.getType());
+        if (isCharacterType(*arg))
+          argTypes.push_back(
+              fir::BoxCharType::get(builder.getContext(), /*FIXME*/ 1));
+        else
+          argTypes.push_back(addr.getType());
         operands.push_back(addr);
```
Comment (jeanPerier) [inline flang/lib/Lower/ConvertExpr.cpp:948]: Shouldn't addr be emboxed right also ?
I think #37 could be reverted if the only reason it was done was to get boxchar in cases like here.
Instead, one could use here:
```
auto char = builder.materializeCharacter(addr);
auto box = builder.createEmboxChar(char.first, char.second);
```
Maybe this pattern should be its own helper function that does nothing if `addr` is already a fir.boxchar.

File: flang/lib/Lower/ConvertExpr.cpp [hunk 2]
```diff
@@ -933,7 +940,11 @@ class ExprLowering {
           addr = builder.createTemporary(getLoc(), val.getType());
           builder.create<fir::StoreOp>(getLoc(), val, addr);
         }
-        argTypes.push_back(addr.getType());
+        if (isCharacterType(*arg))
```
[the remark this answers is not in the document]
Comment (schweitzpgi) [reply flang/lib/Lower/ConvertExpr.cpp:943]: That seems better. I'll give it a shot.

File: flang/lib/Lower/ConvertExpr.cpp (hunk 1 above)
Comment (schweitzpgi) [reply flang/lib/Lower/ConvertExpr.cpp:948]: If I am reading your question right, then a general argument in F77 is not boxed, no. The only descriptor in play is for CHARACTER type arguments. The calling side provides no other information.

Comment (schweitzpgi) [reply flang/lib/Lower/ConvertExpr.cpp:948]: FWIW, #37 was already reverted.

**Pull request merged**